code/switch.js

```diff
@@ -0,0 +1,166 @@
+    "Ok, " +
+      name +
+      " what would you like to order:\n1. Pizza\n2. Milkshake\n3. Salad"
+  );
+  switch (category) {
+    case "1":
+      catName = "Pizza";
+      break;
+    case "2":
+      catName = "Milkshake";
+      break;
+    case "3":
+      catName = "Salad";
+      break;
+    default:
+      alert(
+        "We cannot process orders outside of our categories.\nPlease restart the application."
+      );
+      exit(1);
+  }
+  alert(catName + " is a great choice!");
+
+  // Step 3 - Subtype choice
+  // Your code goes here
+  switch (catName) {
+    case "Pizza":
+      subPizza = prompt(
+        "Pizza? Great choice. Which pizza would you like?\n1. Vesuvio\n2. Margherita\n3. Capricciosa"
+      );
+      switch (subPizza) {
+        case "1":
+          subName = "Vesuvio";
+          break;
+        case "2":
+          subName = "Margherita";
+          break;
+        case "3":
+          subName = "Capricciosa";
+          break;
+        default:
+          alert(
+            "We cannot process orders outside of our categories.\nPlease restart the application."
+          );
+          exit(1);
+      }
+      break;
+    case "Milkshake":
+      subMilkshake = prompt(
+        "Milkshake? Great choice. Which flavour would you like?\n1. Vanilla\n2. Chocolate\n3. Strawberry"
+      );
+      switch (subMilkshake) {
+        case "1":
+          subName = "Vanilla milkshake";
+          break;
+        case "2":
+          subName = "Chocolate milkshake";
+          break;
+        case "3":
+          subName = "Strawberry milkshake";
+          break;
+        default:
+          alert(
+            "We cannot process orders outside of our categories.\nPlease restart the application."
+          );
+          exit(1);
+      }
+      break;
+    case "Salad":
+      subSalad = prompt(
+        "What type of salad would you like?\n1. Ceasar salad\n2. Halloumi salad\n3. Ham and cheese salad"
+      );
+      switch (subSalad) {
+        case "1":
+          subName = "Ceasar salad";
+          break;
+        case "2":
+          subName = "Halloumi salad";
+          break;
+        case "3":
+          subName = "Ham and cheese salad";
+          break;
+        default:
+          alert(
+            "We cannot process orders outside of our categories.\nPlease restart the application."
+          );
+          exit(1);
+      }
+      break;
+    default:
+      alert(
+        "We cannot process orders outside of our categories.\nPlease restart the application."
+      );
+      exit(1);
+  }
+
+  alert(subName + " is a great choice!");
+
+  // Step 4 - Age
+  // Your code goes here
+
+  const age = prompt(
+    name + ", we need to verify your age to determine the portion size:"
+  );
+
+  // Step 5 - Order confirmation
+  // Your code goes here
+  if (age <= priceAge) {
+    portion = "child ";
+    price = "€5";
+  } else {
+    portion = "adult ";
+    price = "€10";
+  }
+
+  const confirm = prompt(
```

Comment:
this is certainly correct, but have you tried using template literals to avoid the string concatenation? Like so:

```
`To confirm your order ${name}, one portion ${subName}, is this correct?
  1. Yes
  2. No`
```